### ISSUE TITLE
`env` JSON Default values not handled when config map missing

### ISSUE THREAD

**Issue by esatterwhite** [razee-io/razeedeploy-core#267]
**Describe the bug**
When using `overrideStrategy: merge` for json keys, the default values are not taken into consideration when a config map is missing rather than the key missing from a config map. Or potentially treated like a scalar value with out being parsed and resetting values merge prior to the missing config map

I think its mostly happening here

https://github.com/razee-io/razeedeploy-core/blob/d821e89d4f858b404212e25010d542481952e0c4/lib/FetchEnvs.js#L65-L71

An error is returned so its never reaching the value casting or the default value logic. To get around this behavior I would have to make sure there is always an empty config map in order to pick up defaults.

**To Reproduce**
```yaml
spec:
  templateEngine: handlebars
  env:
  - name: ingress_rules
    overrideStrategy: merge
    optional: true
    valueFrom:
      configMapKeyRef:
        name: ingress-rules-one
        key: "ingress_rules"
        type: json
    default: |
      '{}'

  # config map is missing
  - name: ingress_rules
    overrideStrategy: merge
    optional: true
    valueFrom:
      configMapKeyRef:
        name: ingress-rules-missing
        key: "ingress_rules"
        type: json
    default: |
      '{"missing": true}'

  - name: ingress_rules
    overrideStrategy: merge
    optional: true
    valueFrom:
      configMapKeyRef:
        name: ingress-rules-three
        key: "ingress_rules"
        type: json
    default: |
      '{}'
```

The end result of this is only what was defined in the config map `ingress-rules-three`. It appears that not only is the missing config map defaults not considered, it is clobbering any values before it.

```yaml
  ingress-rules-three: 1
```

If I remove the definition for the missing config map

```yaml
spec:
  templateEngine: handlebars
  env:
  - name: ingress_rules
    overrideStrategy: merge
    optional: true
    valueFrom:
      configMapKeyRef:
        name: ingress-rules-one
        key: "ingress_rules"
        type: json
    default: |
      '{}'


  - name: ingress_rules
    overrideStrategy: merge
    optional: true
    valueFrom:
      configMapKeyRef:
        name: ingress-rules-three
        key: "ingress_rules"
        type: json
    default: |
      '{}'
```

The end result is a json value of `ingress-rules-one` merged with `ingress-rules-three`

```yaml
  ingress-rules-one: 1
  ingress-rules-three: 1
```

The template I'm testing against just iterates over the merged object and prints the keys

```yaml
  strTemplates:
  - |
    apiVersion: v1
    kind: ConfigMap
    metadata:
      name: test-config
      labels:
        app: "test-config"
        razee/watch-resource: "debug"
    data:
    {{#each ingress_rules}}
      {{{@key}}}: 1
    {{/each}}
```
**Expected behavior**
If a config map is not found, it should use the default value taking into consideration the data type, cast and merge the values

**Kubernetes Version (kubectl version):**
 - Client: 1.18
 - Server: 1.16.9

**Comment by esatterwhite** [comment]
Is it maybe that the default is in quotes?

```yaml
    default: |
      '{}'
```

vs

```yaml
    default: {}
```

**Comment by esatterwhite** [comment]
Oh. I think the quotes may have been the actual problem

**Comment by alewitt2** [comment]
yeah good catch. so basically i think it boils down to desired behavior and where we handle the `default` logic. currently we just use the default value straight up (like you discovered, removing the quotes solved the merge clobbering issue because you are now defining an actual json object that is able to be merged).

so the question becomes, should the default value go through the same parsing the fetched value would have gone through? Because you said the fetched value is type json, should we have parsed the default value as well? Idk, maybe. I could go both ways. I dont think it's a major bug, and i would probably even argue that the type shouldn't apply to the default since type is buried in the `valueFrom` section. If you think its a bug that needs fixing, i would probably lean to giving default its own `type` processing. But my preference would be to leave it as its.

**Comment by esatterwhite** [comment]
Ya, I think the default should be type cast there isn't a great way to understand if the value you are dealing with is the resolved value or if it were coming from a default to be able to do anything special with it. And I really wouldn't want to have to do that anyway. If I'm specifying a type, I expect it to always be that type if it can reasonably be parsed/cast

**Comment by esatterwhite** [comment]
I think this is fix w/ the most recent version.

**Comment by esatterwhite** [comment]
I just ran into this again and forgot that I patched this. Just not on the most recent version.
But the new stuff certainly does type case the defaults

**Comment by esatterwhite** [comment]
Fixed in #293
